### PR TITLE
Initial Docker sls tutorial

### PR DESCRIPTION
**Pull request by thatch45**
### What does this PR do?

Create a small initial tutorial for the feature added in #34484
### What issues does this PR fix or reference?
saltstack/salt#34484

@dmacvicar @isbm This is an initial stab at a tutorial, the only issue I had was picking a good base. The one you used in the PR, `leapy` did not resolve. Also, please let me know what more you think we can add here
@jacobhammons please take a look and clean this up, also can you or someone else give it a spin once we have a solid base image to use?

### CODE REVIEW

**Comment by thatch45** [comment]
Please don't merge this until we figure out which docker base image to use

**Comment by dmacvicar** [comment]
Let me have a discussion about this. I will come back to this PR.

File: doc/topics/tutorials/docker_sls.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    pip install docker-py
+
+Step 2 - Install Salt
+=====================
+
+For this tutorial we will be using Salt Call, which is available in the
+`salt-minion` package, please follow the Salt Installation docs found here:
+https://repo.saltstack.com/
+
+Step 3 - Creat With Salt States
```
Comment (thedrow) [inline doc/topics/tutorials/docker_sls.rst]: Creat? Is this a typo?

Comment (thatch45) [inline doc/topics/tutorials/docker_sls.rst]: typo, I got it

**Comment by thatch45** [comment]
Perfect!

**Comment by thatch45** [comment]
How does that look?

**Comment by dmacvicar** [comment]
"If `base` is omitted the default image used is a minimal openSUSE image with Python support, maintained by SUSE"

**Comment by thatch45** [comment]
Thanks :)
it is never all caps it when talking casually, like in chat, I should so I get used to it

**Comment by dmacvicar** [comment]
I think `ommited` should be `omitted`?

**Comment by thatch45** [comment]
A solid catch! In my defense the vim spell checked does not highlight things in rst notes.

File: doc/topics/tutorials/docker_sls.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Next run the `docker.sls_build` command:
+
+.. code-block:: bash
+
+    salt-call --local dockerng.sls_build test base=my_base_image mods=vim
+
+Now we have a fresh container called `test` to work with and vim has been
+installed.
+
```
Comment (dmacvicar) [inline doc/topics/tutorials/docker_sls.rst]: I think `we have a fresh container` should be `we have a fresh image` as `container` is usually referred to a running instance of an image.

Comment (thatch45) [inline doc/topics/tutorials/docker_sls.rst]: Like that you think?

**Comment by dmacvicar** [comment]
:+1:

File: doc/topics/tutorials/docker_sls.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    vim:
+      pkg.installed
+
+.. note::
+
+    The base image you choose will need to have python 2.6 or 2.7 installed.
+    We are hoping to resolve this constraint in a future release.
+
+    If `base` is omitted the default image used is a minimal OpenSUSE
```
Comment (dmacvicar) [inline doc/topics/tutorials/docker_sls.rst]: OpenSUSE -> openSUSE

Comment (thatch45) [inline doc/topics/tutorials/docker_sls.rst]: LOL, I merged it :)
Ok, I will get it :)

**Comment by dmacvicar** [comment]
made_at_www txt2pic com

**Pull request merged**